Assets/SEE/Utils/ThreadSafeHashSet.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace SEE.Utils
+{
+    /// <summary>
+    /// A thread-safe hash set.
+    /// </summary>
+    /// <typeparam name="T">the element type of the set</typeparam>
+    public class ThreadSafeHashSet<T> : IEnumerable
```

Comment:
I'm not sure, but this may need to implement `IEnumerable<T>` instead, otherwise the generic parameter could be lost for the enumerable methods. (Cf. `System.Collections` vs `System.Collections.Generic`).

Reply:
I fixed it.